agenda/presentation/src/main/java/com/nibalk/tasky/agenda/presentation/detail/DetailState.kt

```diff
@@ -15,6 +17,34 @@ data class DetailState(
     val description: String = "",
     val startDate: LocalDate = LocalDate.now(),
     val startTime: LocalTime = LocalTime.now(),
-    val endDate: LocalDate = LocalDate.now(),
-    val endTime: LocalTime = LocalTime.now(),
-)
+    val details: AgendaItemDetails,
+    val notificationDurationType: NotificationDurationType = NotificationDurationType.THIRTY_MINUTES,
+) {
+    fun <T> getEventDetailField(
+        agendaItemDetails: AgendaItemDetails?,
+        fieldExtractor: (AgendaItemDetails.Event) -> T
+    ): T? {
+        Timber.d("getEventDetailField1 = %s", agendaItemDetails)
+        return if (agendaItemDetails is AgendaItemDetails.Event) {
+            val field = fieldExtractor(agendaItemDetails)
+            Timber.d("getEventDetailField2 = %s", field)
+            return field
+        } else {
+            null
+        }
+    }
+}
+
+sealed interface AgendaItemDetails {
+    data object NoDetails : AgendaItemDetails
```

Comment:
In which scenario would you have no details at all? You always know at the time where you initialize the VM which specific item type you're on. I personally didn't default initialize the details field, but left it up to the ViewModel's initial state, so I can initialize it together with the navigation agenda item type parameter